src/string_with_vars.ml

```diff
@@ -104,21 +103,41 @@ let string_of_var syntax v =
   | Parens -> sprintf "$(%s)" v
   | Braces -> sprintf "${%s}" v
 
+let expand_generic t ~f ~is_multivalued ~to_string =
+  match t.items with
+  | [Var (syntax, v)] when t.unquoted ->
+     (* Unquoted single var *)
+     (match f t.loc v with
+      | Some e -> Inl e
+      | None -> Inr(string_of_var syntax v))
+  | _ ->
+     Inr(List.map t.items ~f:(function
+             | Text s -> s
+             | Var (syntax, v) ->
+                match f t.loc v with
+                | Some x ->
+                   if t.unquoted && is_multivalued x then
+                     raise (Loc.Error (t.loc, "please quote the string \
```

Comment:
Can you use `Loc.fail` instead? Just for uniformity with the rest of the code. I find `Loc.fail` easier to read.

Reply:
OK.